### ISSUE TITLE
Laplacian matrix

### ISSUE THREAD

**Issue by Serendipity-LC** [benedekrozemberczki/GraphWaveletNeuralNetwork#8]
Hi! Why do we need to use all the node data (2707 nodes) when constructing the Laplacian matrix instead of just the training set data?

**Comment by benedekrozemberczki** [comment]
??

**Comment by chenjiakun** [comment]
> Hi! Why do we need to use all the node data (2707 nodes) when constructing the Laplacian matrix instead of just the training set data?

If we do this, do we use the node features of the validation set and test set during the training process?

**Comment by Serendipity-LC** [comment]
> Hi! Why do we need to use all the node data (2707 nodes) when constructing the Laplacian matrix instead of just the training set data?

If we do this, do we use the node features of the validation set and test set during the training process?

**Comment by benedekrozemberczki** [comment]
This is semi supervised.

**Comment by chenjiakun** [comment]
> This is semi supervised.

I know this, but why are the features of the test set used during model training?

**Comment by benedekrozemberczki** [comment]
Please read about semisupervised learning.
